src/lib/api.ts

```diff
@@ -1,27 +1,42 @@
+import type { NetlifyAPI } from 'netlify'
+
 import { warn } from '../utils/command-helpers.js'
-import { SiteInfo } from '../utils/types.js'
+import type { SiteInfo } from '../utils/types.js'
 
-// @ts-expect-error TS(7031) FIXME: Binding element 'api' implicitly has an 'any' type... Remove this comment to see the full error message
-export const cancelDeploy = async ({ api, deployId }) => {
+export const cancelDeploy = async ({ api, deployId }: { api: NetlifyAPI; deployId: string }): Promise<void> => {
   try {
     await api.cancelSiteDeploy({ deploy_id: deployId })
   } catch (error) {
-    // @ts-expect-error TS(2571) FIXME: Object is of type 'unknown'.
-    warn(`Failed canceling deploy with id ${deployId}: ${error.message}`)
+    warn(
+      `Failed canceling deploy with id ${deployId}: ${
+        error instanceof Error ? error.message : error?.toString() ?? ''
+      }`,
+    )
   }
 }
 
 const FIRST_PAGE = 1
 const MAX_PAGES = 10
 const MAX_PER_PAGE = 100
 
-// @ts-expect-error TS(7023) FIXME: 'listSites' implicitly has return type 'any' becau... Remove this comment to see the full error message
-export const listSites = async ({ api, options }): Promise<SiteInfo[]> => {
+export const listSites = async ({
+  api,
+  options,
+}: {
+  api: NetlifyAPI
+  // FIXME(serhalp) `page` and `maxPages` are missing from `netlify` package types
```

Comment:
As discussed with @ndhoule, this is fine for me and pretty limited in scope, but we'll have to pause on adding types to Netlify API stuff in CLI until we solve this properly. Nathan has a bunch of knowledge in this space from doing this in the SDK.